streams/piping/pipe-through.js

```diff
@@ -136,12 +138,37 @@ test(() => {
   };
 
   // An incorrect implementation which uses an internal method to mark the promise as handled will throw or crash here.
-  ReadableStream.prototype.pipeThrough.call(dummy, { });
+  ReadableStream.prototype.pipeThrough.call(dummy, uninterestingReadableWritablePair);
 
   // An incorrect implementation that tries to mark the promise as handled by calling .then() or .catch() on the object
   // will fail these tests.
   assert_false(thenCalled, 'then should not be called');
   assert_false(catchCalled, 'catch should not be called');
 }, 'pipeThrough should not be fooled by an object whose instanceof Promise returns true');
 
+test(() => {
+  const pairs = [
+    {},
+    { readable: undefined, writable: undefined },
+    { readable: 'readable' },
+    { readable: 'readable', writable: undefined },
+    { writable: 'writable' },
+    { readable: undefined, writable: 'writable' }
+  ];
+  for (const pair of pairs) {
+    const rs = new ReadableStream();
+    assert_throws(new TypeError(), () => rs.pipeThrough(pair),
+                  'pipeThrough should throw for argument ' + JSON.stringify(pair) + ';');
+  }
+}, 'undefined readable or writable arguments should cause pipeThrough() to throw');
+
+test(() => {
```

Comment:
I think separate tests for `{ writable: new WritableStream(), readable: arg }` and the counterpart would be more interesting.